ipa-core/src/protocol/ipa_prf/validation_protocol/proof_generation.rs

```diff
@@ -88,32 +103,41 @@ impl ProofBatch {
         let (mut uv_values, first_proof_from_left, my_first_proof_left_share) =
             LargeProofGenerator::gen_artefacts_from_recursive_step(
                 ctx,
-                &mut record_counter,
+                &mut prss_record_ids,
                 &first_lagrange_table,
                 ProofBatch::polynomials_from_inputs(uv_tuple_inputs),
             );
 
-        // approximate length of proof vector (rounded up)
-        let uv_len_bits: u32 = usize::BITS - uv_values.len().leading_zeros();
-        let small_recursion_factor_bits: u32 = usize::BITS - SRF.leading_zeros();
-        let expected_len = 1 << (uv_len_bits - small_recursion_factor_bits);
+        // `MAX_PROOF_RECURSION - 2` because:
+        //  * The first level of recursion has already happened.
+        //  * We need (SRF - 1) at the last level to have room for the masks.
+        let max_uv_values: usize =
+            (SRF - 1) * SRF.pow(u32::try_from(MAX_PROOF_RECURSION - 2).unwrap());
+        tracing::info!("uv_values.len() = {}", uv_values.len());
```

Comment:
trace?

Reply:
I just wanted to see the values in the draft run, I am going to remove this.